code.js

```diff
@@ -1,6 +1,42 @@
+
+
+
+
 document.querySelector("#theTemp").innerHTML = "100";
+let locationText = document.querySelector("#location")
+
+
+// Returns a promise containing the user's latitude and longitude
+function getUserCoordinates() {
+  if (navigator.geolocation) {
+    return new Promise(function (resolve, reject) {
+      navigator.geolocation.getCurrentPosition(function (position) {
+        resolve(success(position));
+      }, reject);
+    });
+  } else {
+    return Promise.reject("Geolocation is not available on this device");
+  }
+
+  function success(position) {
+    return {
```

Comment:
To help other developers, we may want to declare this object type outside your function so that the other devs will know what the return value looks like. Possibly as a "Location" class like this:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes

And then in the comment above say `// Returns a promise containing a Location object`.

But we can always do that later.